### PR TITLE
pm: device: runtime: Do not suspend devices when runtime is enabled

### PR DESCRIPTION
**Pull request by ceolin**
- Add API to check if a device has runtime pm enabled
- Do not suspend a device if it is using device runtime

### CODE REVIEW

File: include/pm/device_runtime.h [hunk 1]
```diff
@@ -116,12 +116,27 @@ int pm_device_runtime_put(const struct device *dev);
...
 int pm_device_runtime_put_async(const struct device *dev);
 
+/**
+ * @brief Check if device runtime is enabled for a given device.
+ *
+ * @funcprops \pre_kernel_ok
+ *
+ * @param dev Device instance.
+ *
+ * @retval true If device has enabled device runtime.
```
Comment (gmarull) [inline include/pm/device_runtime.h:126]: `If the device has device runtime PM enabled`

File: include/pm/device_runtime.h [hunk 2]
```diff
@@ -116,12 +116,27 @@ int pm_device_runtime_put(const struct device *dev);
...
+ * @retval false If device has not enabled device runtime.
```
Comment (gmarull) [inline include/pm/device_runtime.h:127]: `If the device has device runtime PM disabled`

File: subsys/pm/power.c [hunk 3]
```diff
@@ -118,7 +119,8 @@ static int pm_suspend_devices(void)
 		int ret;
 
 		/* ignore busy devices */
```
Comment (gmarull) [inline subsys/pm/power.c:121]: nit: adjust comment here

**Review by gmarull — APPROVED** [review]
LGTM, please adjust docstrings as commented

**Review by teburd — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**